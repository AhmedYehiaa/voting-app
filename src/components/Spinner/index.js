import React from "react";
import styles from "./Spinner.module.css";

const Spinner = () => (
  <div className={styles.spinner}>
    <svg viewBox="0 0 50 50">
      <circle
        className={styles.path}
        cx="25"
        cy="25"
        r="20"
        fill="none"
        strokeWidth="2"
      />
    </svg>
  </div>
);

export default Spinner;
