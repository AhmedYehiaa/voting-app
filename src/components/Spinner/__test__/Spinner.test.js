import React from "react";
import ReactDOM from "react-dom";

import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

import Spinner from "../index";

describe("Spinner Component", () => {
  it("render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Spinner />, div);
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Spinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
