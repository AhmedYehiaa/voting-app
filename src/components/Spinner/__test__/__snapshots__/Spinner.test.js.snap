// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component matches snapshot 1`] = `
<div
  className="spinner"
>
  <svg
    viewBox="0 0 50 50"
  >
    <circle
      className="path"
      cx="25"
      cy="25"
      fill="none"
      r="20"
      strokeWidth="2"
    />
  </svg>
</div>
`;
